src/main/java/uk/ac/cam/cl/dtg/segue/dos/AudienceContext.java

```diff
@@ -0,0 +1,68 @@
+package uk.ac.cam.cl.dtg.segue.dos;
+
+import uk.ac.cam.cl.dtg.isaac.dto.GameFilter;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ *  A class to specify a restricted and *validated* audience context which the user is allowed to specify when creating
+ *  a gameboard.
+ */
+public class AudienceContext {
+    private List<Stage> stage;
+    private List<ExamBoard> examBoard;
+    private List<Difficulty> difficulty;
+    private List<RoleRequirement> role;
+
+    @Nullable
+    public static AudienceContext fromFilter(@Nullable final GameFilter gameFilter) {
+        if (gameFilter == null) {return null;}
+        return new AudienceContext() {{
+            if (gameFilter.getStages() != null) {
+                setStage(gameFilter.getStages().stream().map(Stage::valueOf).collect(Collectors.toList()));
+            }
+            if (gameFilter.getExamBoards() != null) {
+                setExamBoard(gameFilter.getExamBoards().stream().map(ExamBoard::valueOf).collect(Collectors.toList()));
+            }
+            if (gameFilter.getDifficulties() != null) {
+                setDifficulty(gameFilter.getDifficulties().stream().map(Difficulty::valueOf).collect(Collectors.toList()));
+            }
+        }};
+    }
+
+    public AudienceContext() {}
+
+    public List<Stage> getStage() {
+        return stage;
+    }
```

Comment:
A silly decision while creating this structure in the editor was that the field name is singular but the value is a collection...
I am going to figure out if I can do a mass migration of the content to using the pluralised form of the field names before we start saving this in gameboard contexts everywhere but it will require switching off the beta feature for any users with it enabled on live (I believe only staff after a recent clean up) and for content editors to refresh their browsers so that they don't create structures in the old format... It gets a lot harder to do that if I don't manage to fix this before the release.

Reply:
😳